src/components/form/Form.js

```diff
@@ -0,0 +1,204 @@
+import { useState, useEffect, useRef } from 'react';
+import {
+    ConnectForm,
+    Exit,
+    Input,
+    InputWrap,
+    Label, Title,
+    Wrapper,
+    Submit,
+    TextArea,
+    Overlay,
+    Iframe
+} from "./styledForm";
+import exit from '../../images/exit.svg';
+
+const Form = (props) => {
+
+    const formRef = useRef();
+
+    // ________________________________determine form validity
+    const [isValid, setIsValid] = useState(false)
+
+    const [nameValid, setNameValid] = useState(true);
+
+    const [emailValid, setEmailValid] = useState(true);
+
+    // __________________________________inputs and errors
+    const [inputs, setInputs] = useState({
+        name: '',
+        email: '',
+        request: ''
+    })
+
+    const [errors, setErrors] = useState({
+        name: '',
+        email: ''
+    });
+
+    const errorMessage = '(This field is required)'
+
+    const checkFormValidity = () => {
+        setIsValid(formRef.current.checkValidity())
+    }
+
+    // _____________________________ determine input for validation
+    const pickInput = (name, valid) => {
+        if (name === 'name') {
+            if (valid) {
+                setNameValid(true)
+                return
+            }
+            else setNameValid(false)
+        }
+        if (name === 'email') {
+            if (valid) {
+                setEmailValid(true)
+                return
+            }
+            else setEmailValid(false)
+        }
+    }
+
+    const handleChange = (event) => {
+        const { id, value } = event.target;
+        errors[id] && updateErrors(event);
+        setInputs({
+            ...inputs,
+            [id]: value
+        });
+    }
+
+    const updateErrors = (event) => {
+        const { id } = event.target;
+        const valid = event.target.validity.valid;
+        pickInput(id, valid)
+        setErrors({
+            ...errors,
+            [id]: errorMessage
+        });
+    }
+
+    // ________________________________________reset inputs
+    const resetInputs = () => {
+        setInputs({
+            name: '',
+            email: '',
+            request: ''
+        })
+        setErrors({
+            name: '',
+            email: ''
+        })
+        setIsValid(false);
+        setNameValid(true);
+        setEmailValid(true);
+    }
+
+    // ___________________reset inputs when popup is closed
+    useEffect(() => {
+        resetInputs();
+    }, [props.open]);
+
+    // ____________________________close popup
+    const closePopup = () => {
```

Comment:
same comment as previous. You can probably remove this function and by having onClick={props.closePopup} to the close button